File: android/app/src/main/java/com/harshbshah/fer/data/model/RoutineTemplate.kt
package com.harshbshah.fer.data.model

import com.google.firebase.firestore.Exclude
import java.util.Date
import java.util.UUID

/** A saved workout plan (e.g. "Push Day") the user can start from. */
data class RoutineTemplate @JvmOverloads constructor(
    @get:Exclude @set:Exclude
    var id: String? = null,

    var name: String = "",
    // Stored as the iOS SF Symbol name (e.g. "list.bullet.rectangle") so a routine
    // created on either platform renders a matching icon on the other — see
    // RoutineIcons.kt for the SF Symbol -> Material icon mapping used for display.
    var iconName: String = "list.bullet.rectangle",
    var exercises: List<RoutineExercise> = emptyList(),
    var createdAt: Date = Date(),
    var lastUsedAt: Date? = null,
    var ownerId: String = ""
)

data class RoutineExercise @JvmOverloads constructor(
    var id: String = UUID.randomUUID().toString(),
    var exerciseId: String = "",
    var exerciseName: String = "",
    var targetSets: Int = 3,
    var targetReps: Int = 10,
    var restSeconds: Int = 90,
    var notes: String = ""
)

File: Fer/Models/RoutineTemplate.swift
//
//  RoutineTemplate.swift
//  Fer
//
//  A saved workout plan (e.g. "Push Day") the user can start from.
//
//  Plain `id` (not @DocumentID) on purpose — this file is shared with the
//  Watch target, and FirebaseFirestore doesn't support watchOS, so nothing
//  here can import it. FirestoreService assigns `id` manually from the
//  document reference after decoding instead.
//

import Foundation

struct RoutineTemplate: Codable, Identifiable, Hashable {
    var id: String?
    var name: String
    var iconName: String = "list.bullet.rectangle"
    var exercises: [RoutineExercise]
    var createdAt: Date = Date()
    var lastUsedAt: Date? = nil
    var ownerId: String = ""

    static func == (lhs: RoutineTemplate, rhs: RoutineTemplate) -> Bool {
        lhs.id == rhs.id
    }
    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}

struct RoutineExercise: Codable, Identifiable, Hashable {
    var id: String = UUID().uuidString
    var exerciseId: String
    var exerciseName: String
    var targetSets: Int = 3
    var targetReps: Int = 10
    var restSeconds: Int = 90
    var notes: String = ""
}
